src/backend/api/meals.js

```diff
@@ -2,13 +2,76 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+// http://localhost:5000/api/meals	GET,	Returns all meals
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = await knex("meal").select("id", "title");
+
+    meals.length === 0
+    ? res.status(404).send("Meals not found")
+    : res.status(200).send(meals);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+// /api/meals	POST, Adds a new meal to the database
+router.post("/", async (req, res) => {
+  try {
+    const meal = req.body;
+
+    const addMeal = await knex("meal").insert(meal);
+    res.status(201).send({"message": "Added meal"});
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+// /api/meals/:id	GET,	Returns the meal by id
+router.get("/:id", async (req, res) => {
+  try {
+    const id = Number(req.params.id);
+
+    const meal = await knex("meal").select("id", "title").where({id});
+
+    meal.length === 0
+    ? res.status(404).send("Meal not found")
+    : res.status(200).send(meal);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+// /api/meals/:id PUT,	Updates the meal by id
+router.put("/:id", async (req, res) => {
+  try {
+    const id = Number(req.params.id);
```

Comment:
Would prefer to add an if condition for checking id, whether it  is a Number or not. As I can see you used Number() method, it will convert the id to number type but what if the id contains some characters rather than numbers.

Reply:
Good to refactor also in the following endpoints :)